### ISSUE TITLE
Building after dispose when rolling a list causing exceptions

### ISSUE THREAD

**Issue by igorcd** [andyduke/styled_text_package#55]
Hello, first thanks for that magnific package.
My project is throwing the following exception when rolling a ListView (Probabily is building after element hide from list).

```
E/flutter (15254): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: This widget has been unmounted, so the State no longer has a context (and should be considered defunct).
E/flutter (15254): Consider canceling any active work during "dispose" or using the "mounted" getter to determine if the State is still active.

line: package:styled_text/styled_text.dart:341
```

![image](https://user-images.githubusercontent.com/39352365/161140084-7fb553a8-73eb-49b1-940d-7afc39d4b4a0.png)

Thanks

**Comment by andyduke** [comment]
@igorcd Could you provide a minimal code to reproduce the problem?

**Comment by igorcd** [comment]
Example:
```
class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);
  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  var controller = ScrollController();

  void navigateToBottom() {
    WidgetsBinding.instance?.addPostFrameCallback((timeStamp) {
      controller.jumpTo(controller.position.maxScrollExtent);
    });
  }

  @override
  void initState() {
    super.initState();
    navigateToBottom();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
      body: ListView.separated(
        padding: const EdgeInsets.all(20),
        controller: controller,
        itemCount: 500,
        separatorBuilder: (context, index) => const SizedBox(height: 20),
        itemBuilder: ((context, index) {
          return Container(
            padding: const EdgeInsets.all(10),
            decoration: BoxDecoration(
              color: Colors.grey.shade100,
              border: Border.all(color: Colors.grey.shade200),
              borderRadius: BorderRadius.circular(8),
            ),
            child: StyledText(
              text: 'Test: <bold>bold</bold> and <red>red color</red> text.',
              tags: {
                'bold': StyledTextTag(style: const TextStyle(fontWeight: FontWeight.bold)),
                'red': StyledTextTag(style: const TextStyle(fontWeight: FontWeight.bold, color: Colors.red)),
              },
            ),
          );
        }),
      ),
    );
  }
}

```

**Comment by andyduke** [comment]
@igorcd Please check now with version [5.1.0](https://pub.dev/packages/styled_text/versions/5.1.0).

**Comment by igorcd** [comment]
@andyduke resolved in 5.1.0
Thanks for you help.
